x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list_page.test.tsx

```diff
@@ -0,0 +1,284 @@
+      const allFilterOptions = getAllByTestId(`${filterPrefix}-option`);
+
+      const selectedFilterOptions = allFilterOptions.reduce<string[]>((acc, option) => {
+        if (option.getAttribute('aria-checked') === 'true') {
+          acc.push(option.textContent?.split('-')[0].trim() as string);
+        }
+        return acc;
+      }, []);
+
+      expect(selectedFilterOptions.length).toEqual(2);
+      expect(selectedFilterOptions).toEqual(['release', 'processes']);
+      expect(history.location.search).toEqual('?commands=release,processes');
+    });
+
+    it('should read and set status filter values from URL params', () => {
+      const filterPrefix = 'statuses-filter';
+      reactTestingLibrary.act(() => {
+        history.push('/administration/action_history?statuses=pending,failed');
+      });
+
+      render();
+      const { getAllByTestId, getByTestId } = renderResult;
+      userEvent.click(getByTestId(`${testPrefix}-${filterPrefix}-popoverButton`));
+      const allFilterOptions = getAllByTestId(`${filterPrefix}-option`);
+
+      const selectedFilterOptions = allFilterOptions.reduce<string[]>((acc, option) => {
+        if (option.getAttribute('aria-checked') === 'true') {
+          acc.push(option.textContent?.split('-')[0].trim() as string);
+        }
+        return acc;
+      }, []);
+
+      expect(selectedFilterOptions.length).toEqual(2);
+      expect(selectedFilterOptions).toEqual(['Failed', 'Pending']);
+      expect(history.location.search).toEqual('?statuses=pending,failed');
+    });
+
+    // TODO: add tests for hosts and user when those filters are added
+
+    it('should read and set relative date ranges filter values from URL params', () => {
+      reactTestingLibrary.act(() => {
+        history.push('/administration/action_history?startDate=now-23m&endDate=now-1m');
+      });
+
+      render();
+      const { getByTestId } = renderResult;
+
+      expect(getByTestId('superDatePickerstartDatePopoverButton').textContent).toEqual(
+        '~ 23 minutes ago'
+      );
+      expect(getByTestId('superDatePickerendDatePopoverButton').textContent).toEqual(
+        '~ a minute ago'
+      );
+      expect(history.location.search).toEqual('?startDate=now-23m&endDate=now-1m');
+    });
+
+    // FIXME: the UI time should be local time
+    it.skip('should read and set absolute date ranges filter values from URL params', () => {
+      const startDate = '2022-09-12T11:00:00.000Z';
+      const endDate = '2022-09-12T11:30:33.000Z';
+      reactTestingLibrary.act(() => {
+        history.push(`/administration/action_history?startDate=${startDate}&endDate=${endDate}`);
+      });
+
+      const { getByTestId } = render();
+
+      expect(getByTestId('superDatePickerstartDatePopoverButton').textContent).toEqual(
+        'Sep 12, 2022 @ 13:00:00:000'
+      );
+      expect(getByTestId('superDatePickerendDatePopoverButton').textContent).toEqual(
+        'Sep 12, 2022 @ 13:30:33:000'
+      );
```

Comment:
So the date filters work correctly, but here in the test I'm probably missing something that should correctly format the given URL date to the one that's shown in the date filter input. Will get to it tomorrow.

Reply:
Ah okay, I think I need to set a local on the test file so it knows what time to expect.